### PR TITLE
  embeddings: Add Amazon Bedrock embeddings

### PR DESCRIPTION
**Pull request by sansmoraxz**
Adding [AWS bedrock](https://aws.amazon.com/bedrock/) support with embedding generation for cohere and titan.

### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [x] Describes the source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

**Comment by sansmoraxz** [comment]
To test you may use the following:-

```go
package bedrock_test

import (
	"context"
	"fmt"
	"testing"

	. "github.com/tmc/langchaingo/embeddings/bedrock"
)

func TestEmbedQuery(t *testing.T) {
	model, err := NewBedrock(WithModel(MODEL_COHERE_EN))
	if err != nil {
		t.Fatal(err)
	}

	embeddings, err := model.EmbedQuery(context.Background(), "hello world")

	if err != nil {
		t.Fatal(err)
	}

	fmt.Println(len(embeddings))
}

func TestEmbedDocuments(t *testing.T) {
	model, err := NewBedrock(WithModel(MODEL_TITAN_EMBED_G1))
	if err != nil {
		t.Fatal(err)
	}

	embeddings, err := model.EmbedDocuments(context.Background(), []string{"hello world", "goodbye world"})

	if err != nil {
		t.Fatal(err)
	}

	fmt.Println(len(embeddings))
}
```

Have not included in the PR as I saw the API calls were not included in existing test cases.

Might I suggest specialized build tags for API calls?

File: embeddings/bedrock/bedrock_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+	require.NoError(t, err)
+}
+
+func TestEmbedDocuments(t *testing.T) {
+	t.Parallel()
+	model, err := bedrock.NewBedrock(bedrock.WithModel(bedrock.ModelCohereEn))
+	if err != nil {
+		t.Skip("Skipping test, failed to initialize bedrock model")
+	}
```
Comment (sansmoraxz) [inline embeddings/bedrock/bedrock_test.go:28]: This is not working I think, it's picking credentials from somewhere. I need some suggestions. Maybe some new environment variable to be explicitly enabled?

Comment (sansmoraxz) [reply embeddings/bedrock/bedrock_test.go]: Added env `TEST_AWS` if you want to test with AWS default credentials chain.

**Review by tmc — APPROVED** [review]
LGTM

File: embeddings/bedrock/bedrock_test.go (hunk 1 above)
Comment (tmc) [reply embeddings/bedrock/bedrock_test.go]: ehh yeah I'd prefer to check for env var poppulation rather than just skipping on init failure.

**Pull request merged**